### PR TITLE
New docs website / Page "Typography" - Part 2

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

This is a follow-up of #809 to align to the most recent designs.

### :hammer_and_wrench: Detailed description

In this PR I have:
- refactored the `Doc::FontHelpersList` component so can be used in a more general way
- updated the `Typography` page to match as much as possible the designs
- added a small fix for the text indentation in the "sidecar" element

👉 👉 👉 **Preview**: https://hds-website-git-new-docs-website-typography-part2-hashicorp.vercel.app/foundations/typography/

@cveigt this is the best I could achieve, for different technical reasons. I have collated all the markdown in a single page, instead of sections, per designs. But I haven't actually looked or reviewed the content itself, this will be done later by someone else.

### :link: External links

Jira ticket: https://hashicorp.atlassian.net/browse/HDS-1260
Figma file: https://www.figma.com/file/Ky0qWjvHZR3je1lCBlzNB1/HDS-website?node-id=1131%3A67683

***

### 👀 Reviewer's checklist:

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: website/docs/foundations/typography/index.md [hunk 1]
```diff
@@ -2,13 +2,79 @@
...
-  @include "partials/code/how-to-use.md"
-  @include "partials/code/showcase.md"
-</section>
+Each OS comes with a specific typeface. By telling our components “use the system’s default typeface, whatever that is,” we’re giving our apps fewer things to have to look for, so they load faster. This is important in big apps, where you need every little bit of performance help you can get. This of course, means that it might appear a little different on each OS, but contextually to that user on that OS, it will not seem out of place at all.
 
+System fonts also offer many benefits that many web fonts don’t. They are broadly tested and have many styles and variations to support internationalization, code, tabular data, data viz, etc... As a new design system, we need to be pragmatic, and system fonts seem to cover our present and potential future use cases.
+
+!!! Info
+
+HashiCorp has general writing guidelines for copy standards e.g. capitalization and punctuation. Refer to [this document](https://docs.google.com/document/d/1ABNkYIp6aLwPYNpuYOFcDFBFCcnH_4ePRaLP-gD_tKE/edit#heading=h.w9d2qpiz1vpb) for more details.
```
Comment (alex-ju) [inline website/docs/foundations/typography/index.md:13]:
```suggestion
HashiCorp has [general writing guidelines](https://docs.google.com/document/d/1ABNkYIp6aLwPYNpuYOFcDFBFCcnH_4ePRaLP-gD_tKE/edit#heading=h.w9d2qpiz1vpb) for copy standards e.g. capitalization and punctuation.
```

**Review by alex-ju — APPROVED** [review]
Looks great! I left a small suggestion where we can cut some content and make the link text more relevant.

File: website/docs/foundations/typography/partials/code/how-to-use.md [hunk 2]
```diff
@@ -51,10 +51,14 @@ These are the **CSS helper classes** that you can use:
...
 To use this classes you have to import the CSS file `[products|devdot]/css/helpers/typography.css` from the `@hashicorp/design-system-tokens` package.
 
-!!! info
+#### Combinations font style / font weight
 
-**Important**
+!!! Info
 
-While the _font-family/font-weight/typography_ helpers can be combined together in code, in reality not all the combinations are valid from the design perspective: please refer to the design documentation to see which styles combinations are allowed.
+**Notice**
```
Comment (MelSumner) [inline website/docs/foundations/typography/partials/code/how-to-use.md:58]:
```suggestion
**Note**
```

**Review by cveigt — APPROVED** [review]
It look good. Thanks for making these updates quickly 🚀

**Pull request merged**